firebase-firestore/src/androidTest/java/com/google/firebase/firestore/TransactionTest.java

```diff
@@ -709,23 +709,23 @@ public void testDoesNotRetryOnPermanentError() {
   public void testRetryOnAlreadyExistsError() {
     final FirebaseFirestore firestore = testFirestore();
     DocumentReference doc = firestore.collection("foo").document();
-    CountDownLatch latch = new CountDownLatch(2);
+    AtomicInteger retryCount = new AtomicInteger(0);
     waitFor(
         firestore.runTransaction(
             transaction -> {
-              latch.countDown();
+              retryCount.getAndIncrement();
               transaction.get(doc);
               // Do a write outside of the transaction.
-              waitFor(doc.set(map("count", 1)));
-              // Now try to set the doc within the transaction.This should fail once
+              if (retryCount.get() == 1) waitFor(doc.set(map("count", retryCount.get())));
+              // Now try to set the doc within the transaction. This should fail once
               // with ALREADY_EXISTS error.
-              transaction.set(doc, map("count", 2));
+              transaction.set(doc, map("count", 22));
```

Comment:
Instead of setting `map("count", 22)` here, can you set a different, completely unrelated field? Otherwise, it incorrectly implies that it is somehow related. But, in fact, it doesn't matter. e.g. `map("foo", "bar")`.